src/components/modelsList/ModelsList.tsx

```diff
@@ -0,0 +1,146 @@
+import React, { useState, useEffect } from 'react';
+import {
+  Dialog,
+  DialogTitle,
+  DialogContent,
+  DialogActions,
+  List,
+  ListItem,
+  ListItemText,
+  CircularProgress,
+  Box,
+} from '@mui/material';
+import { addModel } from '../../redux/slices/scene';
+import {
+  fetchGLBUrl,
+  fetchModelsList,
+  fetchModelColors,
+} from '../../utils/firebaseUtils';
+import { useAppDispatch } from '../../redux';
+import NotificationPopup, {
+  initialSnackBarState,
+  SnackBarState,
+} from '../notificationPopup/NotificationPopup';
+import TextButton from '../TextButton/TextButton';
+import './ModelsList.css';
+
+interface modelItem {
+  id: string;
+  name: string;
+}
+
+const ModelsList = (): JSX.Element => {
+  const dispatch = useAppDispatch();
+  const [modelsList, setModelsList] = useState<modelItem[]>([]);
+  const [selectedModel, setSelectedModel] = useState<modelItem | null>(null);
+  const [colors, setColors] = useState<string[]>([]);
+  const [loadingColors, setLoadingColors] = useState(false);
+  const [openColorDialog, setOpenColorDialog] = useState(false);
+  const [snackbar, setSnackbar] = useState<SnackBarState>(initialSnackBarState);
+
+  useEffect(() => {
+    const fetchModels = async () => {
+      try {
+        const modelsListFetched = await fetchModelsList();
+        setModelsList(modelsListFetched);
+      } catch (error) {
+        console.error('Error fetching models:', error);
+      }
+    };
+    fetchModels();
+  }, []);
+
+  const handleAddModel = async (model: modelItem, color: string) => {
+    try {
+      const url = await fetchGLBUrl(model.id, color);
+      dispatch(addModel(model.id, model.name, color, url));
+      setSnackbar({
+        open: true,
+        message: `Added ${model.name} with color ${color}`,
+        severity: 'success',
+      });
```

Comment:
we're using this a lot of times, with only couple of types for `severity` and a changing message, would it make sense if this was a util function with the message as a parameter?